### ISSUE TITLE
Ensuring site name uniqueness

### ISSUE THREAD

**Issue by ltseng** [techmatters/terraso-product#190]
## User Story

- As a user, I cannot create a site that shares the same name as another site I am a member of within the same app. 
- As a user, I can create a site with the same name as another site across different apps - i.e. US vs. international
- As a user, I can be added to a project that contains sites that share names with other sites I am a member of.

### Design
- [ ] https://github.com/techmatters/terraso-product/issues/351

### Front end
- [ ] https://github.com/techmatters/terraso-mobile-client/issues/372

### Back end
- [ ] https://github.com/techmatters/terraso-backend/issues/765
- [ ] https://github.com/techmatters/terraso-backend/issues/766
- [ ] https://github.com/techmatters/terraso-backend/issues/767

## Acceptance criteria

- I cannot create a site with the same name as another individual (i.e. non-affiliated site) I am a member of.
- I can create a site with the same name as long as they use different soil information sources (i.e. US vs. international)
- I can create a site with the same name as a site that belongs to a project.

## Resources
[Creating a site](https://docs.google.com/document/d/1vxITdvD9EOjOa92BG7iaSyv55iWZpNCKPDlJOh4sb2Q/edit#heading=h.2ojb3y3asxe4)

**Comment by david-code** [comment]
@ltseng Can a user create a site in a project that has the same name as a site already present in the project?

**Comment by ltseng** [comment]
Yes, as long as they are not already part of an unaffiliated site with the same name. 

(I understand this introduces additional edge cases where a person could create a site named "Site A," add it to a project, and then create another site called "Site A." My hope in that case is that people could distinguish between the two sites by looking for the one that is either part of or not part of a project)

**Comment by CourtneyLee333** [comment]
@ltseng This is a lot of nuance to display in an error message to the user trying to give their new site a name. What is the distilled version that will work? "This name is taken"?

**Comment by ltseng** [comment]
I think "This name is taken" should be sufficient. I wish this were simpler, to be honest 😅
